lib/model-definition.js

```diff
@@ -302,3 +302,25 @@ ModelDefinition.prototype.toJSON = function (forceRebuild) {
   this.json = json;
   return json;
 };
+
+/**
+ * Checks if the id property has been defined in the model definition. Returns
+ * true if the id is found in the model definition properties OR in at least one
+ * id object from the _id property.
+ *
+ * @returns {Boolean} True if id is defined, false otherwise
+ */
+ModelDefinition.prototype.hasIdDefined = function() {
+  var isDefined = false;
+
+  if (this.properties.id)
+    isDefined = true;
+
+  // check if id is defined by a relation
+  this.ids().forEach(function(item) {
+    if (item.hasOwnProperty('id'))
```

Comment:
The check for `false` happens before this. If the model definition has `id: false`, I believe `this.properties.id` is not set at all. Same goes for `this.ids()`.